test/uk/gov/hmrc/helptosavefrontend/connectors/EmailVerificationConnectorSpec.scala

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2017 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.helptosavefrontend.connectors
+
+import org.joda.time.Period
+import play.api.Configuration
+import play.api.http.Status
+import play.api.libs.json.{JsValue, Writes}
+import uk.gov.hmrc.helptosavefrontend.TestSupport
+import uk.gov.hmrc.helptosavefrontend.models.VerifyEmailStatus
+import uk.gov.hmrc.play.http.ws._
+import uk.gov.hmrc.play.http.{HeaderCarrier, HttpReads, HttpResponse}
+import uk.gov.hmrc.play.test.UnitSpec
+
+import scala.concurrent.Future
+
+class EmailVerificationConnectorSpec extends UnitSpec with TestSupport
+{
+
+  val nino = "AE123XXXX"
+  val email = "email@gmail.com"
+
+  def postHttpMock(returnedStatus: Int, returnedData: Option[JsValue]): WSHttp = {
+    val mockHttp = mock[WSHttp]
```

Comment:
don't need to create a mock everytime the function is called - create one once outside of the def